capstone/fitness/src/main/java/com/google/sps/servlets/CreateWorkoutServlet.java

```diff
@@ -0,0 +1,82 @@
+    
+    // Get the JSON string that we created in JavaScript from the request's body
+    String requestBody = request.getReader().readLine();
+    JSONObject workoutJSON = new JSONObject(requestBody);
+    
+    Entity user = DataHandler.getUser();
+
+    // Get the user's current workout list
+    ArrayList<String> userWorkouts = gson.fromJson(DataHandler.getUserData(DataHandler.WORKOUT_LIST_PROPERTY,
+                                                                           user), new TypeToken<List<String>>(){}.getType());
+
+    // Add this new workout to their list
+    String workoutName = (String) workoutJSON.get(DataHandler.WORKOUT_NAME_PROPERTY); 
+    userWorkouts.add(workoutName);
+    user.setProperty(DataHandler.WORKOUT_LIST_PROPERTY, gson.toJson(userWorkouts));
+
+    String workoutType = (String) workoutJSON.get(DataHandler.TYPE_PROPERTY);
+    int weeksToTrain = Integer.parseInt((String) workoutJSON.get(DataHandler.WEEKS_TO_TRAIN_PROPERTY));
+
+    Entity newWorkout = new Entity(DataHandler.WORKOUT_ENTITY, workoutName);
+    // Set the properties that all workouts share
+    newWorkout.setProperty(DataHandler.PROGRESS_PROPERTY, "[]");
+    newWorkout.setProperty(DataHandler.GOAL_STEPS_PROPERTY, "[]");
+    newWorkout.setProperty(DataHandler.TYPE_PROPERTY, workoutName);
+    newWorkout.setProperty(DataHandler.WEEKS_TO_TRAIN_PROPERTY, weeksToTrain);
+
+    // Assign the correct properties depending on the type
+    if(workoutType.equals("lifting")) {
+      newWorkout.setProperty(DataHandler.INITIAL_WEIGHT_PROPERTY, 
+                             Float.parseFloat((String) workoutJSON.get(DataHandler.INITIAL_WEIGHT_PROPERTY)));
+      newWorkout.setProperty(DataHandler.INITIAL_REPS_PROPERTY, 
+                             Integer.parseInt((String) workoutJSON.get(DataHandler.INITIAL_REPS_PROPERTY)));
+      newWorkout.setProperty(DataHandler.GOAL_WEIGHT_PROPERTY, 
+                             Float.parseFloat((String) workoutJSON.get(DataHandler.GOAL_WEIGHT_PROPERTY)));
+      newWorkout.setProperty(DataHandler.GOAL_REPS_PROPERTY, 
+                             Integer.parseInt((String) workoutJSON.get(DataHandler.GOAL_REPS_PROPERTY)));
+    }
+    else {
+      newWorkout.setProperty(DataHandler.MARATHON_LENGTH_PROPERTY, 
+                             Float.parseFloat((String) workoutJSON.get(DataHandler.MARATHON_LENGTH_PROPERTY)));
+      newWorkout.setProperty(DataHandler.INITIAL_TIME_PROPERTY, 
+                             Float.parseFloat((String) workoutJSON.get(DataHandler.INITIAL_TIME_PROPERTY)));
+      newWorkout.setProperty(DataHandler.GOAL_TIME_PROPERTY, 
+                             Float.parseFloat((String) workoutJSON.get(DataHandler.GOAL_TIME_PROPERTY)));
+      newWorkout.setProperty(DataHandler.MILE_TIME_PROPERTY, 
+                             Float.parseFloat((String) workoutJSON.get(DataHandler.MILE_TIME_PROPERTY)));
+    }
```

Comment:
Consider adding a check for unknown workout types just in case.

Reply:
The workoutType comes from a drop-down box so it can only be those two values.